### ISSUE TITLE
Sort out yum repos on pLinux and zLinux boxes

### ISSUE THREAD

**Issue by sxa555** [AdoptOpenJDK/openjdk-infrastructure#714]
docker-ce repo doesn't work on several p/z-Linux, EPEL failed on zLinux - details in https://github.com/AdoptOpenJDK/openjdk-build/issues/938

Machines have been reset to the "failure" state so probably won't install anything at the moment. Will need appropriate playbook adjustments which this issue can cover as I want to close the immediate build one.

**Comment by sxa555** [comment]
docker-ce seems to have been added on `ppc64le` and `s390x` under https://github.com/AdoptOpenJDK/openjdk-infrastructure/pull/702 - @HusainYusufali @jdekonin Can you investigate / resolve please?

**Comment by sxa555** [comment]
Kicked off another run via ansible on `build-marist-rhel74-s390x-2` but it failed as the docker-ce repo is already configured on the system.

Have issued: `rm -f /etc/yum.repos.d/docker-ce.repo`

**Comment by sxa555** [comment]
Also renamed `/etc/yum.repos.d/epel*repo` as the update was failing:

```
[root@build-marist-rhel74-s390x-2 yum.repos.d]# yum update
Loaded plugins: langpacks, product-id, search-disabled-repos, subscription-
              : manager
This system is not registered with an entitlement server. You can use subscription-manager to register.
epel/s390x/metalink                                     |  50 kB     00:00     
Could not parse metalink https://mirrors.fedoraproject.org/metalink?repo=epel-7&arch=s390x error was 
No repomd file
```

**Comment by sxa555** [comment]
Now running through on build-marist-s390x-1 ... Had an issue with disk space, and have symlinked /usr/local/gcc to another location under /data but AWX has been able to complete the playbook execution there now

**Comment by sxa555** [comment]
Playbook run successfully. Now doing the same on build-marist-rhel74-s390x-1 and removing docker-ce from build-osuosl-centos74-ppc64le-2 (but retaining epel on there as it's not causing a problem). No comparable updates were needed on build-osuosl-centos74-ppc64le-1 as it didn't have the `docker-ce.repo` in place (although it currently has a `docker.repo` pointing at http://ftp.unicamp.br/pub/ppc64el/rhel/7/docker-ppc64el which may need investigated/addressed )

**Comment by sxa555** [comment]
On the two zLinux marist boxes I have issued the following:
```
mv /var/cache/yum /data/var-cache-yum
ln -s /data/var-cache-yum /var/cache/yum
```
This should help with the problems related to the relatively small root file system on those systems as seen in https://github.com/AdoptOpenJDK/openjdk-infrastructure/issues/557 and https://github.com/AdoptOpenJDK/openjdk-infrastructure/issues/481 and ensure that the updates are applied in a timely manner by the cron entry going forward (This may need adding to the playbooks ...)

**Comment by sxa555** [comment]
ppc64le playbooks ran through ok. Now validating marist 1

**Comment by sxa555** [comment]
Spotted that https://github.com/AdoptOpenJDK/openjdk-infrastructure/pull/691 isn't yet merged so despite the correct level being on the box the playbooks were attempting to put 7.3 on. This is causing some problems because the `/usr/local/gcc -> /data/usr-local-gcc` link (which we should formalise similar to the `var-cache-yum` one is being overridden with a real directory on the `/` file system by the gcc 7.3.0 tarball extraction. Temporarily patched by symlnking gcc-7.4->gcc-7.3 to stop the playbook from re-extracting 7.3 until the PR is merged.

**Comment by sxa555** [comment]
Completed successfully on all ppc64le and s390x build systems. Closing.
